### DISCUSSION TITLE
ONNX ops for training with PyTorch and TensorFlow

### DISCUSSION THREAD

**Discussion by kiszk** [microsoft/onnxruntime#13336] [Other Q&A]
I am excited to see activities to enable model training on ONNX. How is ONNX going to co-operate with other AI frameworks (e.g. PyTorch, TensorFlow) in a context of training? The current ORT seems to convert an inference graph from PyTorch to a training one by adding operations, such as loss functions, backward computations, and optimizers, as described in [this presentation](https://wiki.lfaidata.foundation/pages/viewpage.action?pageId=35160391&preview=/35160391/39092346/ONNX%20Runtime%20Training%20Update%20-%20Workshop%20March%202021.pdf).
Is this approach along with a long-term plan for ORT? Or is ORT expected to consume a training graph from PyTorch or TensorFlow, for forward/backward/update, using operations in [com.microsoft domain ops](http://www.xavierdupre.fr/app/mlprodict/helpsphinx/onnxops/onnx_commicrosoft_AdamOptimizer.html) for training?

Please let me know if there is a better forum to discuss this topic.
Thank you so much.
